firmware/Sources/MID/MID_DABS/mid_dabs.c

```diff
@@ -0,0 +1,292 @@
+ * @param errors, give access to the errors register to now if there are any error.
+ * @param mode, Output of the function to give the mode for the leds.
+ * @return @ref MID_DABS_RESULT_SUCCESS if all data gathered correctly,
+ * 		@ref MID_DABS_RESULT_BUSY, @ref MID_DABS_RESULT_TIMEOUT or
+ * 		@ref MID_DABS_RESULT_ERROR otherwise.
+ */
+static MID_DABS_result_e CheckBlinkStatus(MID_REG_mode_e epcmode, int16_t lscurr,
+		MID_REG_errorStatus_s * errors, blink_conf_s * mode){
+	MID_DABS_result_e res = MID_DABS_RESULT_SUCCESS;
+	if (errors->hsVoltErr !=MID_REG_ERROR_NONE){
+		*mode = errHsvoltMode;
+	}
+	else if (errors->lsVoltErr !=MID_REG_ERROR_NONE){
+		*mode = errlsvoltMode;
+	}
+	else if (errors->lsCurrErr !=MID_REG_ERROR_NONE){
+		*mode = errlscurrMode;
+	}
+	else if (errors->commErr !=MID_REG_ERROR_NONE){
+		*mode = errCommMode;
+	}
+	else if (errors->tempErr !=MID_REG_ERROR_NONE){
+		*mode = errTempMode;
+	}
+	else if (errors->intErr !=MID_REG_ERROR_NONE){
+		*mode = errIntMode;
+	}
+	// After checking there are no errors, check the actual mode.
+	else{
+		if (epcmode == MID_REG_MODE_IDLE){
+			*mode = idleMode;
+		}
+		else if (epcmode == MID_REG_MODE_WAIT){
+			*mode = waitMode;
+		}
+		else if (epcmode == MID_REG_MODE_CV && lscurr>=0){
+			*mode = cvChgMode;
+		}
+		else if (epcmode == MID_REG_MODE_CV && lscurr<0){
+			*mode = cvDchgMode;
+		}
+		else if (epcmode == MID_REG_MODE_CC && lscurr>=0){
+			*mode = ccChgMode;
+		}
+		else if (epcmode == MID_REG_MODE_CC && lscurr<0){
+			*mode = ccDchgMode;
+		}
+		else if (epcmode == MID_REG_MODE_CP && lscurr>=0){
+			*mode = cpChgMode;
+		}
+		else if (epcmode == MID_REG_MODE_CP && lscurr<0){
+			*mode = cpDchgMode;
+		}
+		else {
+			res = MID_DABS_RESULT_ERROR;
+		}
+	}
+	return res;
+}
+
+/**
+ * @fn MID_DABS_result_t SetLeds()
+ * @brief Set the leds to the state indicated by the input
+ * @param state, 8bit input but only needed the last 4 bits as there are only 4 leds
+ * @return @ref MID_DABS_RESULT_SUCCESS if measured correctly,
+ * 		@ref MID_DABS_RESULT_BUSY, @ref MID_DABS_RESULT_TIMEOUT or
+ * 		@ref MID_DABS_RESULT_ERROR otherwise.
+ */
+static MID_DABS_result_e SetLeds(uint8_t step){
+	MID_DABS_result_e res = MID_DABS_RESULT_ERROR;
+	uint8_t state = ledsMode.mode.ledsStep[step];
+	uint8_t mask= 0x1;
+	res = (MID_DABS_result_e) HAL_GpioSet(HAL_GPIO_OUT_Led0, state & mask);
+	res = (MID_DABS_result_e) HAL_GpioSet(HAL_GPIO_OUT_Led1, (state>>1) & mask);
+	res = (MID_DABS_result_e) HAL_GpioSet(HAL_GPIO_OUT_Led2, (state>>2) & mask);
+	res = (MID_DABS_result_e) HAL_GpioSet(HAL_GPIO_OUT_Led3, (state>>3) & mask);
+	return res;
+}
+
+/**********************************************************************************/
+/*                        Definition of exported functions                        */
+/**********************************************************************************/
+
+//MID_DABS_result_e MID_DabsUpdateMeas(const MID_DABS_meas_e type, MID_REG_meas_s * measreg){
```

Comment:
commented??

Reply:
Not working yet, expecting hal_adc operational to test it

Reply:
Now working perfectly. Solved the problem with a led in the evaluation board.